codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -320,6 +328,24 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
 		}
 	}
 
+	{{if .genDuckLogic}}
```

Comment:
the code will get here for deleted resources as well and it is not allowed to update generation for a deleted object